cmd/sysext.go

```diff
@@ -0,0 +1,139 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/gofrs/uuid"
+	"github.com/kairos-io/enki/pkg/config"
+	"github.com/kairos-io/kairos-sdk/sysext"
+	"github.com/kairos-io/kairos-sdk/utils"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+func NewSysextCmd() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "sysext NAME CONTAINER",
+		Short: "Generate a sysextension from the last layer of the given CONTAINER",
+		Args:  cobra.ExactArgs(2),
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			arch := viper.GetString("arch")
+			if arch != "amd64" && arch != "arm64" {
+				return fmt.Errorf("unsupported architecture: %s", arch)
+			}
+			return nil
+		},
+		RunE: func(cobraCmd *cobra.Command, args []string) error {
+			// Set this after parsing of the flags, so it fails on parsing and prints usage properly
+			cobraCmd.SilenceUsage = true
+			// we log the errors with our nice logger so stop cobra from logging them, just let it return the exit codes
+			cobraCmd.SilenceErrors = true
+
+			cfg, err := config.ReadConfigBuild(viper.GetString("config-dir"), cobraCmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			name := args[0]
+			_, err = os.Stat(fmt.Sprintf("%s.sysext.raw", name))
+			if err == nil {
+				_ = os.Remove(fmt.Sprintf("%s.sysext.raw", name))
+			}
+			cfg.Logger.Logger.Info().Msg("🚀 Start sysext creation")
+
+			dir, _ := os.MkdirTemp("", "")
+			defer func(path string) {
+				err := os.RemoveAll(path)
+				if err != nil {
+					cfg.Logger.Logger.Error().Str("dir", dir).Err(err).Msg("⛔ removing dir")
+				}
+			}(dir)
+			cfg.Logger.Logger.Debug().Str("dir", dir).Msg("creating directory")
+			// Get the image struct
+			cfg.Logger.Logger.Info().Msg("💿 Getting image info")
+
+			platform := fmt.Sprintf("linux/%s", viper.Get("arch"))
+			image, err := utils.GetImage(args[1], platform, nil, nil)
+			if err != nil {
+				cfg.Logger.Logger.Error().Str("image", args[1]).Err(err).Msg("⛔ getting image")
+				return err
+			}
+			// Only for sysext, confext not supported yet
+			AllowList := regexp.MustCompile(`^usr|^/usr`)
```

Comment:
same question here than in the other pr about the dir name

Reply:
done!